libs/matdbg/src/ShaderReplacer.cpp

```diff
@@ -300,11 +300,11 @@ void ShaderIndex::encodeShadersToIndices() {
         offset += sizeof(uint32_t);
 
         const char* s = record.decodedShaderText.c_str();
-        size_t length = record.decodedShaderText.length();
-        for (size_t cur = 0; s[cur] != '\0'; cur++) {
+        const size_t length = record.decodedShaderText.length();
+        for (size_t cur = 0; cur < length; cur++) {
```

Comment:
is it possible to have a `\0` before we reach `length`?

Reply:
This is not possible because length() counts up to the null.

Btw I just noticed that we are already inconsistent about using length() vs size()  (they are equivalent)